### ISSUE TITLE
Canada Post cannot retrieve rates for destinations outside of Canada

### ISSUE THREAD

**Issue by amphibian** [verbb/postie#84]
When using the Canada Post integration, rates cannot be retrieved for destinations outside of Canada. This is because Postie will always send the destination to Canada Post's API using the `<domestic>` node, as opposed to the `<united-states>` or `<international>` nodes as needed. Thus a US destination returns an error from Canada Post such as:

> element `postal-code` is not a valid instance of type `PostalCodeType`

(This is because it's actually a US zip code.)

**Fix**

In the Canada Post provider file, replace the contents of the `<destination>` node with something like this:

```
if($order->shippingAddress->countryIso == 'CA') {
    $payload .= '<domestic>
        <postal-code>' . $orderZipCode . '</postal-code>
    </domestic>';
} elseif($order->shippingAddress->countryIso == 'US') {
    $payload .= '<united-states>
        <zip-code>' . $orderZipCode . '</zip-code>
    </united-states>';
} else {
    $payload .= '<international>
        <country-code>' . $order->shippingAddress->countryIso . '</country-code>
    </international>';
}
```

**Additional info**
- Plugin version: 2.4.15
- Craft version: 3.7.30.1
- Multi-site: No

**Comment by engram-design** [comment]
Completely overlooked this for international shipments at the time, thanks for that. I've just added/fixed this for the next version. To get the fix early, change your `verbb/postie` requirement in `composer.json` to:

```
"require": {
  "verbb/postie": "dev-craft-3 as 2.4.15",
  "...": "..."
}
```

Then run `composer update`.

**Comment by amphibian** [comment]
Thanks Josh, appreciated!

**Comment by engram-design** [comment]
Released in [2.4.16](https://github.com/verbb/postie/releases/tag/2.4.16)
